examples/sagaMonitor/modules/Manager.js

```diff
@@ -0,0 +1,30 @@
+/** The manager is used for bookkeeping all the effect descriptors */
+export default class Manager {
+  constructor() {
+    this.rootIds = []
+    // effect-id-to-effect-descriptor
+    this.map = {}
+  }
+
+  get(effectId) {
+    return this.map[effectId]
+  }
+
+  set(effectId, desc) {
+    this.map[effectId] = desc
+  }
+
+  setAsRoot(effectId) {
+    this.rootIds.push(effectId)
+  }
+
+  getRootIds() {
+    return this.rootIds
+  }
+
+  getChildIds(parentEffectId) {
```

Comment:
seems like we should bookkeep the tree structure here too for perf reasons

Reply:
Yes. Keep an map from parent to children would enhance the performance.